ies-route53/providers/record.rb

```diff
@@ -0,0 +1,70 @@
+def aws
+  {
+    :provider => 'AWS',
+    :aws_access_key_id => new_resource.aws_access_key_id,
+    :aws_secret_access_key => new_resource.aws_secret_access_key
+  }
+end
+
+def zone_id
+  @zone_id ||= new_resource.zone_id
+end
+
+def name
+  @name ||= begin
+    return new_resource.name + '.' if new_resource.name !~ /\.$/
+    new_resource.name
+  end
+end
+
+def value
+  @value ||= new_resource.value
+end
+
+def type
+  @type ||= new_resource.type
+end
+
+def ttl
+  @ttl ||= new_resource.ttl
+end
+
+def overwrite
+  @overwrite ||= new_resource.overwrite
+end
+
+action :create do
+  require 'aws-sdk'
```

Comment:
Maybe also add a Exception handler around this with a Chef Error if the gem cant be required.